### PR TITLE
🌱 Drop cgroup-driver patch for CAPD

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR is a follow up of https://github.com/kubernetes-sigs/cluster-api/pull/8881 automatic generation of another kubeletExtraArg and dropping corresponding code from templates all around.

The PS also includes the cleanup of the kubeletExtraArgs from the cmd/clusterctl/client/cluster/assets folder, which I forgot in the previous PR

/cc @sbueringer @killianmuldoon

### CODE REVIEW

File: test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml [hunk 1]
```diff
@@ -126,44 +126,6 @@ spec:
         valueFrom:
           template: |
             imageRepository: {{ .lbImageRepository }}
```
Comment (sbueringer) [inline test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml:128]: Should we now also be able to drop the corresponding handling for MachinePools in the upgrades-cgroupfs cluster-template?  (folder cluster-template-upgrades-cgroupfs + the if/else in cluster_upgrade_test.go)

**Comment by sbueringer** [comment]
I would wait with merging until tomorrow. Just to avoid interfering with a clean signal for the release today.

I would cherry-pick this into 1.4. So that we don't have two different ways of handling this that we have to maintain / potentially debug in the future (1.3 will basically go away with the 1.5 release)

**Review by killianmuldoon — COMMENTED** [review]
This is an awesome cleanup! 

/lgtm
/approve

**Comment by sbueringer** [comment]
> I would wait with merging until tomorrow. Just to avoid interfering with a clean signal for the release today.
> 
> I would cherry-pick this into 1.4. So that we don't have two different ways of handling this that we have to maintain / potentially debug in the future (1.3 will basically go away with the 1.5 release)

Just more context on that. The test which is affected by this change cannot be run on a PR. That's why I would defer to tomorrow to not break periodics today (+ my open comment)

File: test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml (hunk 1 above)
Comment (fabriziopandini) [reply test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml:128]: I can test this, I just need to catch up with this edge case

Comment (sbueringer) [reply test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml:128]: tl;dr is basically that we have this because MachinePools are not supported in ClusterClass, so we needed more than the ClusterClass to set cgroup driver correctly

Comment (sbueringer) [reply test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml:128]: Looks good!

**Comment by sbueringer** [comment]
Change looks good

/lgtm
/approve

Would probably still wait until after the release with hold cancel, but up to you

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

Agreed that we should hold this until after todays release, given the unforseen issues we've had with some of the changes to CAPD in the last couple of weeks.

**Comment by fabriziopandini** [comment]
squashed, ok to wait

**Comment by sbueringer** [comment]
Should we cherry-pick?
